features/nanostack/FEATURE_NANOSTACK/mbed-mesh-api/mbed-mesh-api/ThreadInterface.h

```diff
@@ -34,6 +34,23 @@ class ThreadInterface : public MeshInterfaceNanostack {
     ThreadInterface(NanostackRfPhy *phy) : MeshInterfaceNanostack(phy) { }
 
     nsapi_error_t initialize(NanostackRfPhy *phy);
+
+    /**
+     * \brief Sets the eui64 parameter for the device config
+     * \return MESH_ERROR_NONE on success.
+     * \return MESH_ERROR_PARAM in case of illegal parameters.
+     * */
+    mesh_error_t device_eui64_set(uint8_t *eui64);
```

Comment:
Needs to be a const pointer. Don't see the point of an error return, if it's just a null check. How can an eui64 be invalid? I guess you could fault multicast ones?

Reply:
Giving NULL is the only case of an error. This is an application api, so even a simple NULL check should be done?  Could be done with assert?

Reply:
I'm just generally very down on spending code space specifically on NULL checks in APIs. If someone somehow does it, it'll crash. There are lots of other invalid things - pointers to uninitialised memory, pointers to already-freed blocks, pointers outside RAM, multicast addresses...  You can't totally protect people from themselves.

With the error test, you'll return the error, and they just won't bother checking the return code anyway...

Given that the existing calls in this API don't bother with NULL checks, I don't see a reason to start now.

Anyway, minor style nit - not really that bothered. The code-in-assert thing below is critical though.